internal/entities/handlers/handler.go

```diff
@@ -130,11 +130,45 @@ func (b *handleEntityAndDoBase) handleRefreshEntityAndDo(msg *watermill.Message)
 
 func (b *handleEntityAndDoBase) forwardEntityCheck(
 	ctx context.Context,
+	entMsg *message.HandleEntityAndDoMessage,
 	ewp *models.EntityWithProperties) error {
 	if ewp == nil {
 		return nil
 	}
 
+	if err := b.matchPropertiesCheck(entMsg, ewp); err != nil {
+		return err
+	}
+
+	return b.repoPrivateOrArchivedCheck(ctx, ewp)
+}
+
+func (_ *handleEntityAndDoBase) matchPropertiesCheck(
```

Comment:
Can you add more documentation on why we need to do this matching? e.g. a case that represents the use of this. I'm afraid this will not be easy to follow or use in the future.

Reply:
done!